dts/common/yaml/usb.yaml

```diff
@@ -0,0 +1,38 @@
+---
+title: USB Base Structure
+id: USB
+version: 0.1
+
+description: >
+    This binding gives the base structures for all USB devices
+
+properties:
+  - compatible:
+      type: string
+      category: required
+      description: compatible strings
+
+  - clock-frequency :
+      type: int
+      category: optional
+      description: Initial clock frequency in Hz
+      generation: define
+
+  - clocks:
+      type: array
+      category: required
+      description: Clock gate information
+      generation: define
+
+  - endpoints-number:
+      type: int
+      category: required
+      description: Number of bidirectional endpoints (including EP0)
```

Comment:
is there something specific about them being bi-directional?

Reply:
bidirectional is not necessary here. Regardless of the USB driver, USB stack can/should check if there are enough endpoints available.